geoips/commandline/commandline_interface.py

```diff
@@ -83,10 +85,90 @@ def execute_command(self):
             sys.exit(2)
 
 
+def support_legacy_procflows():
+    """Run a series of checks on sys.argv to support legacy calls to run_procflow.
+
+    The new GeoIPS CLI has changed how we run our process workflows from now on. Instead
+    of calling 'run_procflow' or 'data_fusion_procflow', we now call
+    'geoips run single_source' or 'geoips run data_fusion'. This function parses through
+    sys.argv and performs the necessary translations to match our current format so that
+    the CLI's argparser can call the appropriate functionality.
+
+    Returns
+    -------
+    legacy: bool
+        - The truth value as to whether or not a legacy procflow call was used
+    """
+    defined_procflow = None
+    supported_procflows = ["config_based", "data_fusion", "single_source"]
+    if (
+        basename(sys.argv[0]) == "geoips"
+        and sys.argv[1] == "run"
+        and (len(sys.argv) < 3 or sys.argv[2] not in supported_procflows)
+    ):
+        # Either a procflow was not specified or it was an invalid procflow. Notify the
+        # user of that with a 'NotImplementedError'.
+        if len(sys.argv) >= 3:
+            procflow_name = sys.argv[2]
+        else:
+            procflow_name = "No procflow supplied."
+        raise NotImplementedError(
+            f"'geoips run' was called alongside procflow: '{procflow_name}'.\nIf you "
+            "did not supply a procflow name, this is not supported currently.\n"
+            "Eventually, 'geoips run' will call the 'order_based' procflow, however "
+            "this is not at the current time of use.\nFor a list of supported "
+            f"procflows, choose one of the following: {supported_procflows}."
+        )
+    elif basename(sys.argv[0]) == "run_procflow":
+        entrypoint = "run_procflow"
+        defined_procflow = "single_source"
+
+    elif basename(sys.argv[0]) == "data_fusion_procflow":
+        entrypoint = "data_fusion_procflow"
+        defined_procflow = "data_fusion"
+
+    if defined_procflow:
+        # Either 'run_procflow' or 'data_fusion_procflow' was called, make the
+        # appropriate translations in sys.argv so the CLI's parser can call the correct
+        # functionality
+        if "--procflow" in sys.argv:
+            # If '--procflow' was found in the command line arguments, loop through the
+            # arguments and grab the correct procflow.
+            for idx, arg in enumerate(sys.argv):
+                if arg == "--procflow" and len(sys.argv) > idx + 1:
+                    defined_procflow = sys.argv[idx + 1]
+        sys.argv[0] = sys.argv[0].replace(entrypoint, "geoips")
+        sys.argv.insert(1, "run")
+        sys.argv.insert(2, defined_procflow)
```

Comment:
Ahh, so this is just converting the explicit data_fusion_procflow call to run geoips data_fusion ?  Works for me.

Reply:
Yup! A little hacky but it works for the time being.